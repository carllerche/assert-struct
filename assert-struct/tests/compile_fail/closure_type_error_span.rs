use assert_struct::assert_struct;

#[derive(Debug)]
struct Container {
    value: String,
}

fn main() {
    let container = Container { 
        value: "test".to_string() 
    };
    
    // This should fail with a type error pointing to the specific closure parameter
    assert_struct!(container, Container {
        value: |x: i32| x > 0,  // Type error: closure expects i32 but gets &String
        ..
    });
}