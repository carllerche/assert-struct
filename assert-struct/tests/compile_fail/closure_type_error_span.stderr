error[E0631]: type mismatch in closure arguments
  --> tests/compile_fail/closure_type_error_span.rs:15:16
   |
15 |         value: |x: i32| x > 0,  // Type error: closure expects i32 but gets &String
   |                ^-------
   |                |
   |                expected due to this
   |                found signature defined here
   |
   = note: expected closure signature `fn(&String) -> _`
              found closure signature `fn(i32) -> _`
note: required by a bound in `check_closure_condition`
  --> src/lib.rs
   |
   |     pub fn check_closure_condition<T, F>(value: T, predicate: F) -> bool
   |            ----------------------- required by a bound in this function
   |     where
   |         F: FnOnce(T) -> bool,
   |            ^^^^^^^^^^^^^^^^^ required by this bound in `check_closure_condition`
